develop/items/custom-enchantment-effects.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Custom Enchantment Effects
+description: Learn how to create your enchantment effects.
+authors:
+  - krizh-p
+---
+
+# Custom Enchantments {#custom-enchantments}
+
+Since 1.21, custom enchantments have taken a "data-driven" approach; while this makes adding simple enchantments (ie. increasing attack damage) more straightforward and easy, it also complicates creating complex enchantments. This was done by breaking down what enchantments do into effect components. An effect component contains code for what special things an enchantment should do. By default, Minecraft supports various effects such as item damage, knockback, experience, and more--however, this guide will focus on creating custom enchantment effects that are not supported by default. **[Before contuining, it is heavily suggested you first determine if the default Minecraft effects will work for your usecase.](https://minecraft.wiki/w/Enchantment_definition#Effect_components)** The rest of the guide will assume you understand how "simple" data-driven enchantments are configured.
+
+# Custom Enchantment Effects {#custom-enchantment-effects}
+
+After you've confirmed that your enchantment will not work with the default effect components, we can start by creating an `enchantment` folder, and within it create a folder `effect`. Within the `effect` folder, we'll create the `LightningEnchantmentEffect` record.
+
+Next, we can create a constructor and override the `EnchantmentEntityEffect` interface methods. We'll also create `CODEC` variable to encode and decode our effect; A `MapCodec` is a part of Minecraft's serialization system, which allows for converting data between different formats (e.g., from Java objects to JSON or binary data).   
```

Comment:
No need for the info on MapCodec, you can just do something like "For more information on Codecs, consider having a read of the [Codecs page]\(...)